clang/test/SemaSYCL/undefined-functor.cpp

```diff
@@ -0,0 +1,19 @@
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -sycl-std=2020 -verify -fsyntax-only %s
+// This test checks that an error is thrown when a functor without a call operator defined is used as a kernel.
+
+#include "sycl.hpp"
+
+using namespace sycl;
+queue q;
+
+struct FunctorWithoutCallOperator; // expected-note {{forward declaration of 'FunctorWithoutCallOperator'}}
```

Comment:
Can you add another variant case where the struct is defined but without the call operator?